packages/sdks/overrides/angular/src/components/dynamic-renderer/dynamic-renderer.ts

```diff
@@ -88,160 +81,2537 @@ export default class DynamicRenderer {
+      );
+    }
+  }
+
+  ngOnDestroy() {
+    this._listenerFns.forEach((fn) => fn());
+  }
+}
+
+@Component({
+  selector: 'dynamic-canvas, DynamicCanvas',
+  template: ` <canvas #v><ng-content></ng-content></canvas> `,
+  standalone: true,
+  styles: [':host { display: contents; }'],
+})
+export class DynamicCanvas {
+  @Input() attributes!: any;
+  @Input() actionAttributes?: any;
+
+  @ViewChild('v', { read: ElementRef }) v!: ElementRef;
+
+  _listenerFns = new Map<string, () => void>();
+
+  constructor(private renderer: Renderer2) {}
+
+  setAttributes(el: any, value: any, changes?: any) {
+    if (!el) return;
+
+    const target = changes ? changes : value;
+    Object.keys(target).forEach((key) => {
+      if (key.startsWith('on')) {
+        if (this._listenerFns.has(key)) {
+          this._listenerFns.get(key)!();
+        }
+        this._listenerFns.set(
+          key,
+          this.renderer.listen(
+            el,
+            key.replace('on', '').toLowerCase(),
+            target[key]
+          )
+        );
+      } else {
+        this.renderer.setAttribute(el, key, target[key] ?? '');
+      }
+    });
+  }
+
+  ngAfterViewInit() {
+    this.setAttributes(this.v?.nativeElement, this.attributes);
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.attributes) {
+      this.setAttributes(
+        this.v?.nativeElement,
+        this.attributes,
+        changes.attributes.currentValue
+      );
+    }
+  }
+
+  ngOnDestroy() {
+    this._listenerFns.forEach((fn) => fn());
+  }
+}
+
+@Component({
+  selector: 'dynamic-iframe, DynamicIframe',
+  template: ` <iframe #v><ng-content></ng-content></iframe> `,
+  standalone: true,
+  styles: [':host { display: contents; }'],
+})
+export class DynamicIframe {
+  @Input() attributes!: any;
+  @Input() actionAttributes?: any;
+
+  @ViewChild('v', { read: ElementRef }) v!: ElementRef;
+
+  _listenerFns = new Map<string, () => void>();
+
+  constructor(private renderer: Renderer2) {}
+
+  setAttributes(el: any, value: any, changes?: any) {
+    if (!el) return;
+
+    const target = changes ? changes : value;
+    Object.keys(target).forEach((key) => {
+      if (key.startsWith('on')) {
+        if (this._listenerFns.has(key)) {
+          this._listenerFns.get(key)!();
+        }
+        this._listenerFns.set(
+          key,
+          this.renderer.listen(
+            el,
+            key.replace('on', '').toLowerCase(),
+            target[key]
+          )
+        );
+      } else {
+        this.renderer.setAttribute(el, key, target[key] ?? '');
+      }
+    });
+  }
+
+  ngAfterViewInit() {
+    this.setAttributes(this.v?.nativeElement, this.attributes);
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.attributes) {
+      this.setAttributes(
+        this.v?.nativeElement,
+        this.attributes,
+        changes.attributes.currentValue
+      );
+    }
+  }
+
+  ngOnDestroy() {
+    this._listenerFns.forEach((fn) => fn());
+  }
```

Comment:
haha this is wild 😂. 

would it be possible to write a script that generates these? The same way we have `build-inline-fns` running after Mitosis but before `ng build`. we could have a `generate-dynamic-renderer`.

It could use a template like:

```ts
@Component({
  selector: '${SELECTOR}',
  template: ` <${ELEMENT_NAME} #v><ng-content></ng-content></${ELEMENT_NAME}> `,
  standalone: true,
  styles: [':host { display: contents; }'],
})
export class DynamicIframe {
  @Input() attributes!: any;
  @Input() actionAttributes?: any;

  @ViewChild('v', { read: ElementRef }) v!: ElementRef;

  _listenerFns = new Map<string, () => void>();

  constructor(private renderer: Renderer2) {}

  setAttributes(el: any, value: any, changes?: any) {
    if (!el) return;

    const target = changes ? changes : value;
    Object.keys(target).forEach((key) => {
      if (key.startsWith('on')) {
        if (this._listenerFns.has(key)) {
          this._listenerFns.get(key)!();
        }
        this._listenerFns.set(
          key,
          this.renderer.listen(
            el,
            key.replace('on', '').toLowerCase(),
            target[key]
          )
        );
      } else {
        this.renderer.setAttribute(el, key, target[key] ?? '');
      }
    });
  }

  ngAfterViewInit() {
    this.setAttributes(this.v?.nativeElement, this.attributes);
  }

  ngOnChanges(changes: SimpleChanges) {
    if (changes.attributes) {
      this.setAttributes(
        this.v?.nativeElement,
        this.attributes,
        changes.attributes.currentValue
      );
    }
  }

  ngOnDestroy() {
    this._listenerFns.forEach((fn) => fn());
  }
}
```

and append the code to `dynamic-renderer.ts` looping over an array of HTML tags.

that will make it a lot easier to maintain this override in the future.